pennylane/templates/subroutines/prepselprep.py

```diff
@@ -0,0 +1,200 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the PrepSelPrep template.
+This template contains a decomposition for performing a block-encoding on a
+linear combination of unitaries using the Prepare, Select, Prepare method.
+"""
+# pylint: disable=arguments-differ,import-outside-toplevel,too-many-arguments
+import copy
+
+import numpy as np
+
+import pennylane as qml
+from pennylane.operation import Operation
+
+
+def _get_new_terms(lcu):
+    """Compute a new sum of unitaries with positive coefficients"""
+
+    coeffs, _ = lcu.terms()
+
+    if qml.math.iscomplexobj(coeffs):
+        new_coeffs, new_ops = _new_terms_is_complex(lcu)
+    else:
+        new_coeffs, new_ops = _new_terms_is_real(lcu)
+
+    interface = qml.math.get_interface(lcu.terms()[0])
+    new_coeffs = qml.math.array(new_coeffs, like=interface)
+
+    return new_coeffs, new_ops
```

Comment:
Hi @willjmax , a final small suggestion. If you update the function this way, you can remove `_new_terms_is_complex` and `_new_terms_is_real`.
This will compact the code, but also will solve a  issue that arose when using complex numbers. Currently, operators are being duplicated and this is forcing you to use more control qubits than necessary.

```python
def _get_new_terms(lcu):
    """Compute a new sum of unitaries with positive coefficients"""

    new_coeffs = []
    new_ops = []

    for coeff, op in zip(*lcu.terms()):

        angle = qml.math.angle(coeff)
        new_coeffs.append(qml.math.abs(coeff))

        new_op = op @ qml.GlobalPhase(-angle, wires=op.wires)
        new_ops.append(new_op)

    interface = qml.math.get_interface(lcu.terms()[0])
    new_coeffs = qml.math.array(new_coeffs, like=interface)

    return new_coeffs, new_ops
```

Reply:
Thanks, this was a great suggestion. I just pushed the change.